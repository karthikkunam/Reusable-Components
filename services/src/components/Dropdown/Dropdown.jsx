import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Dropdown = props => {
  const [showDropdown, setShowDropdown] = useState(false);
  const [dropdownValue, setDropdownValue] = useState(
    props.value ? (props.value.label) : (props.placeHolder ? props.placeHolder : (props.options && props.options[0].label)),
  );
  const { id, options, ariaExpanded, ariaHaspopup, dataToggle, onClick, disabled, className } = props;

  const handleDropdown = () => {
    setShowDropdown(!showDropdown);
  };

  const onDropdownSelect = option => {
    onClick && onClick(option);
    setDropdownValue(option.label);
    handleDropdown();
  };

  return (
    <div className={`component-example mb-3 col-sm-12 ${className}`}>
      <div className={`dropdown ${showDropdown ? 'show' : ''}`}>
        <button
          className="btn btn-primary dropdown-toggle"
          onClick={handleDropdown}
          type="button"
          id={id}
          data-toggle={dataToggle}
          aria-haspopup={ariaHaspopup}
          disabled={disabled}
          aria-expanded={ariaExpanded}>
          {dropdownValue}
          {options && options.length > 0 && <span className="icon icon-angle-down"></span>}
        </button>
        <div className={`dropdown-menu ${showDropdown ? 'show' : ''}`} aria-labelledby="dropdownMenuButton">
          {options &&
            options.length > 0 &&
            options.map((option, index) => {
              return (
                <a key={index} className="dropdown-item" onClick={() => onDropdownSelect(option)}>
                  {option.label}
                </a>
              );
            })}
        </div>
      </div>
    </div>
  );
};
Dropdown.defaultProps = {
  id: 'dropdownMenuButton',
  ariaExpanded: 'false',
  ariaHaspopup: 'true',
  dataToggle: 'dropdown',
};

Dropdown.propTypes = {
  id: PropTypes.string,
  options: PropTypes.arrayOf({}).isRequired,
  value: PropTypes.string.isRequired,
  ariaExpanded: PropTypes.string,
  ariaHaspopup: PropTypes.string,
  dataToggle: PropTypes.string,
  placeHolder: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  className: PropTypes.string
};

export default Dropdown;
