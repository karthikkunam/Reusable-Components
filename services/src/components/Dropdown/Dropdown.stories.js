import React from 'react';
import Dropdown from '.';

export default {
  title: 'Dropdown',
  component: Dropdown,
};

const options = [
  {
    label: 'Option 1',
    value: 1,
  },
  {
    label: 'Option 2',
    value: 2,
  },
  {
    label: 'Option 3',
    value: 3,
  },
];
const defaultOptions = [
  {
    label: 'Select an option',
    value: 'none',
  },
  {
    label: 'Option 1',
    value: 1,
  },
  {
    label: 'Option 2',
    value: 2,
  },
  {
    label: 'Option 3',
    value: 3,
  },
];
const value = {
  label: 'No options available',
  value: 'none',
};
const placeHolder = 'It is a placeholder';

export const DropdownWithPlaceholder = () => <Dropdown placeHolder={placeHolder} options={options} />;
export const DropdownWithOptions = () => <Dropdown options={defaultOptions} />;
export const DropdownWithoutOptions = () => <Dropdown value={value} />;
export const DisabledDropdown = () => <Dropdown disabled={true} options={defaultOptions} />;
