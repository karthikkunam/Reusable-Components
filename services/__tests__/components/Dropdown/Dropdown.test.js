import React from 'react';
import { shallow } from 'enzyme';
import Dropdown from '../../../src/components/Dropdown';

const defaultProps = {
  id: 'dropdownMenuButton',
  options: [{}],
  ariaExpanded: 'false',
  ariaHaspopup: 'true',
  dataToggle: 'dropdown',
  disabled: false,
};
describe('Dropdown Component', () => {
  it('renders correctly', () => {
    const component = shallow(<Dropdown {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  it('renders Dropdown component with placeholder', () => {
    const props = {
      placeHolder: 'It is a placeholder',
    };
    const component = shallow(<Dropdown {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders Dropdown component with options', () => {
    const props = {
      defaultOptions: [
        {
          label: 'Select an option',
          value: 'none',
        },
        {
          label: 'Option 1',
          value: 1,
        },
        {
          label: 'Option 2',
          value: 2,
        },
        {
          label: 'Option 3',
          value: 3,
        },
      ],
    };
    const component = shallow(<Dropdown {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders default value in Dropdown', () => {
    const props = {
      value: [
        {
          label: 'No options available',
          value: 'none',
        },
      ],
    };
    const component = shallow(<Dropdown {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('renders disabled Dropdown', () => {
    const props = {
      disabled: true,
    };
    const component = shallow(<Dropdown {...defaultProps} {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('Should call handleDropdown function', () => {
    const props = {
      setShowDropdown: jest.fn(),
    };
    const component = shallow(<Dropdown {...defaultProps} {...props} />);
    component
      .find('button')
      .at(0)
      .simulate('click', {
        handleDropdown: () => {},
      });
    expect(component).toMatchSnapshot();
  });
  it('Should call onDropdownSelect function', () => {
    const props = {
      onClick: jest.fn(),
      setDropdownValue: jest.fn(),
      handleDropdown: jest.fn(),
    };
    const component = shallow(<Dropdown {...defaultProps} {...props} />);
    component
      .find('a')
      .at(0)
      .simulate('click', {
        onDropdownSelect: () => {},
      });
    expect(component).toMatchSnapshot();
  });
});
