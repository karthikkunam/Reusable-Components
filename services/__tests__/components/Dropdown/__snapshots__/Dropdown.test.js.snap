// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Component Should call handleDropdown function 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown show"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={false}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu show"
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown Component Should call onDropdownSelect function 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown show"
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={false}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu show"
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown Component renders Dropdown component with options 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown "
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={false}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu "
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown Component renders Dropdown component with placeholder 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown "
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={false}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      It is a placeholder
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu "
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown Component renders correctly 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown "
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={false}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu "
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown Component renders default value in Dropdown 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown "
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={false}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu "
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;

exports[`Dropdown Component renders disabled Dropdown 1`] = `
<div
  className="component-example mb-3 col-sm-12 undefined"
>
  <div
    className="dropdown "
  >
    <button
      aria-expanded="false"
      aria-haspopup="true"
      className="btn btn-primary dropdown-toggle"
      data-toggle="dropdown"
      disabled={true}
      id="dropdownMenuButton"
      onClick={[Function]}
      type="button"
    >
      <span
        className="icon icon-angle-down"
      />
    </button>
    <div
      aria-labelledby="dropdownMenuButton"
      className="dropdown-menu "
    >
      <a
        className="dropdown-item"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
